src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1161,9 +1160,8 @@ private SolutionCompilationState ComputeFrozenSnapshot(CancellationToken cancell
                     oldDocumentState != newDocumentState &&
```

Comment:
>oldDocumentState != newDocumentState [](http://example.com/codeflow?start=20&length=36)

unrelated: This part of the condition confuses me. Is the "oldDocumentState != newDocumentState" part of the condition necessary?

Reply:
It will commonly be equal for most of the documents in a frozen project that is currently being worked with. So this allows us to avoid doing any checking by just doing a pointer check.